BMS_Sender/BMS_DataSender.cpp

```diff
@@ -0,0 +1,78 @@
+
+#define CATCH_CONFIG_MAIN
+
+#include "../test/catch.hpp"
+#include "BMS_DataSender.h"
+#include "../receiver/BMS_receiver.h"
+
+TEST_CASE("validate the Read operation function - readFileData_e") {
+	
+	StreamOpSts_t streamOpSts_s;
+	char filePath[100] = "./BMS_Sender/SensorData.txt";
+	
+	streamOpSts_s.readSts_e   = OP_MODE_STATUS_INIT;
+        streamOpSts_s.streamSts_e = OP_MODE_STATUS_INIT;
+	streamOpSts_s.streamedDataLen_i = 0;
+	
+	streamOpSts_s = readFileData_e(filePath);
+	
+  	REQUIRE(streamOpSts_s.readSts_e   == OP_MODE_STATUS_SUCCESSFUL);
+	REQUIRE(streamOpSts_s.streamSts_e   == OP_MODE_STATUS_SUCCESSFUL);
+	REQUIRE(streamOpSts_s.streamedDataLen_i   == 50);
+}
+
+TEST_CASE("validate the Read operation function with Invalid File path - readFileData_e") {
+	
+	StreamOpSts_t streamOpSts_s;
+	char filePath[100];
+	
+	streamOpSts_s.readSts_e   = OP_MODE_STATUS_INIT;
+        streamOpSts_s.streamSts_e = OP_MODE_STATUS_INIT;
+	streamOpSts_s.streamedDataLen_i = 0;
+	
+	streamOpSts_s = readFileData_e(filePath);
+	
+	
+  	REQUIRE(streamOpSts_s.readSts_e   == OP_MODE_STATUS_FAILED);
+	REQUIRE(streamOpSts_s.streamSts_e   == OP_MODE_STATUS_INIT);
+	REQUIRE(streamOpSts_s.streamedDataLen_i   == 0);
+}
+
+TEST_CASE("validate the Send operation to console function - sendDataToConsole_e") {
```

Comment:
Overall a good combination of positive and negative tests.
However, the crucial functionality of the sender is the output format. If this format is not explicitly tested, future maintainers may change it and only realize while integrating with the receiver. That would be quite late, in comparison with a simple unit test.
Also, parsing your own output will indicate how easy it would be for a consumer of your output. That is the purpose of writing tests - to look at it from a consumer's point of view.